File: src/main/kotlin/com/smnsyh/hr/controller/ApiController.kt
package com.smnsyh.hr.controller

abstract class ApiController {
    companion object {
        private const val API_PATH = "/api/v1"

        const val AUTHENTICATE_URL: String = "$API_PATH/auth"
        const val SYSTEM_USER_URL: String = "$API_PATH/user"
        const val SYSTEM_ROLE_URL: String = "$API_PATH/role"
        const val SYSTEM_POSITION_URL: String = "$API_PATH/position"

        // Spring Boot Actuator services
        const val AUTOCONFIG_ENDPOINT = "/autoconfig"
        const val BEANS_ENDPOINT = "/beans"
        const val CONFIGPROPS_ENDPOINT = "/configprops"
        const val ENV_ENDPOINT = "/env"
        const val MAPPINGS_ENDPOINT = "/mappings"
        const val METRICS_ENDPOINT = "/metrics"
        const val SHUTDOWN_ENDPOINT = "/shutdown"
    }
}

File: frontend/src/app/shared/api.controller.ts
export class ApiController {

  public static readonly BACKEND_URL = "http://localhost:8080/";
  public static readonly API_BASE = "api/v1";

  public static readonly AUTH_URL = ApiController.BACKEND_URL + ApiController.API_BASE + "/auth";
  public static readonly USER_API_URL = ApiController.BACKEND_URL + ApiController.API_BASE + "/user";
  public static readonly ROLE_API_URL = ApiController.BACKEND_URL + ApiController.API_BASE + "/role";
  public static readonly POSITION_API_URL = ApiController.BACKEND_URL + ApiController.API_BASE + "/position";


}
